### DISCUSSION TITLE
Como faz para receber dados de uma API?

### DISCUSSION THREAD

**Discussion by endersonmenezes** [codaqui/institucional#68] [Q&A] [answered]
Quem consegue me dar um código para realizar receber os dados dessa API: `https://jsonplaceholder.typicode.com/users` e transformar isso em uma lista, ou uma tabela?

Será que algum de vocês podem me ajudar: @codaqui/membros =]

**Comment by LuizFernando-TC** [comment]
Esse link tem uma com tabela:
https://pt.stackoverflow.com/questions/408504/renderizar-dados-json-recebidos-de-uma-api-em-uma-tabela-html-django

**Reply by endersonmenezes** [reply]
Cola o código aqui e explica, usando a API que dei de exemplo, me ajuda kkkkk ! =]

Se olha como fica o seguinte texto:
```python
print("Hello World!")
```

![image](https://user-images.githubusercontent.com/11020807/195893418-852d6170-108e-4e8f-8038-fa83f0eb389a.png)

**Comment by ArtumosGOC** [comment, marked as answer]
Dei uma pesquisada na internet e consegui encontrar algumas coisas que me auxiliaram a criar esse código em JS para receber dados da API, não conheço muito as sintaxes usadas no código, tentei usar o getJson, mas não consegui.

*Java Script:*
```javascript
var i=0;
var items = [];
fetch("https://jsonplaceholder.typicode.com/users").then(resposta =>
{
   return resposta.json();
}).then(elementos =>
{

   console.log(elementos);
   for(i;i <= elementos.length;i++)
   {
      if(i == 10)
      {
         $( "<ul/>", { "class": "list", html: items.join( "" ) } ).appendTo( "body" );
      }
      else
      {
         items.push( "<li id='" , elementos[i].id, "'>","ID: ", elementos[i].id," Nome: ", elementos[i].name," Nome de Usuario: ", elementos[i].username ,"</li>" );
      }
   }
   

})
```

*HTML:*

```html
<!DOCTYPE html>
<html>
<head>
<title>JSON</title>
    <script 
    src="https://code.jquery.com/jquery-3.2.1.min.js">
    </script>
    <script src="getapi.js"></script>
</head>

<body>
</body>
</html>
```

O uso do `<script  src="https://code.jquery.com/jquery-3.2.1.min.js"></script>` é essencial para o funcionamento.


Resultado: 

![image](https://user-images.githubusercontent.com/63540372/195963473-e244e4db-c469-42ba-870d-ef3262b2fdb5.png)

`links de ref: `

- [jQuery.getJSON](https://api.jquery.com/jquery.getjson/)
- [JavaScript read JSON from URL](https://zetcode.com/javascript/jsonurl/)
- [como consumir uma api](https://www.youtube.com/watch?v=IDG6EOXYAq8)

**Reply by ArtumosGOC** [reply]
@endersonmenezes

**Reply by ArtumosGOC** [reply]
Getapi.js é o código de cima.
elementos.length pode ser usada no if.

**Reply by endersonmenezes** [reply]
Muito bom, visto que eu não disse uma linguagem. Hoje o jQuery não é muito convencionado para o uso em grande escala, as grandes empresas começaram a utilizar outras ferramentas para isso, consegue pesquisar duas e citar aqui? Não precisa do código de exemplo dai o/

**Reply by endersonmenezes** [reply]
(Editei o seu código fazendo com que você não precise colocar a imagem para "pintar" o código, veja as edições que eu fiz)
